common/background_exception.h

```diff
@@ -0,0 +1,51 @@
+#pragma once
+
+// Standard C++ includes
+#include <stdexcept>
+
+namespace BoBRobotics {
+//----------------------------------------------------------------------------
+// BoBRobotics::BackgroundException
+//----------------------------------------------------------------------------
+//! A wrapper for passing exceptions between threads (i.e. a background thread to the main one)
+class BackgroundException {
+public:
+    //! Start catching exceptions on background threads
+    static void enableCatching()
+    {
+        m_DoCatch = true;
+    }
+
+    //! Sets the global exception
+    static void set(const std::exception_ptr &error)
+    {
+        if (m_DoCatch) {
+            m_Exception = error;
+        } else {
+            std::rethrow_exception(error);
+        }
+    }
+
+    //! Checks if the global exception is set and rethrows it if so
+    static void check()
+    {
+        if (m_Exception) {
+            std::rethrow_exception(m_Exception);
+        }
+    }
+
+private:
+    static bool m_DoCatch;
+    static std::exception_ptr m_Exception;
+}; // BackgroundException
+
+/*
+ * We want to *not* include definitions when we're linking against another
+ * object file. This currently only seems to be a problem when linking with
+ * libbebop.
+ */
+#ifndef NO_BACKGROUND_EXCEPTION_DEFINITIONS
```

Comment:
Yep! I know initialising static variables in headers is a no-no, but I thought it would be annoying to have to include these definitions in almost every .cc file we have. So I've just changed libbebop to make with ``-DNO_BACKGROUND_EXCEPTION_DEFINITIONS`` and this kinda gets round it. We'll just have to remember to do this for any other lib*s that we do in future which also use this class.